src/jit/emitarm64.cpp

```diff
@@ -4132,23 +4132,13 @@ void emitter::emitIns_R_R(
 
         case INS_fcvtl:
         case INS_fcvtl2:
-            assert(isVectorRegister(reg1));
-            assert(isVectorRegister(reg2));
-            assert(isValidVectorDatasize(size));
-            assert(isValidArrangement(size, opt));
-            elemsize = optGetElemsize(opt);
-            assert(elemsize == EA_4BYTE); // Widening from Float to Double, opt should correspond to src layout
-            fmt = IF_DV_2G;
-            break;
-
         case INS_fcvtn:
         case INS_fcvtn2:
             assert(isVectorRegister(reg1));
             assert(isVectorRegister(reg2));
             assert(isValidVectorDatasize(size));
-            assert(isValidArrangement(size, opt));
-            elemsize = optGetElemsize(opt);
-            assert(elemsize == EA_8BYTE); // Narrowing from Double to Float, opt should correspond to src layout
+            assert(insOptsNone(opt));
+            assert(size == EA_8BYTE); // Narrowing from Double or Widening to Double (Half not supported)
```

Comment:
What does "Half not supported" mean?

Reply:
ARM64 has support for Double (64-bit), Single (32 bit), and Half (16-bit) floating point sizes.  CoreCLR does not use Half sized floating point.  It is the only other size used for these instructions.

Reply:
Based on the interest shown in https://github.com/dotnet/coreclr/issues/11948 can this be implemented for aarch64 and other architectures later?

Reply:
@kasper3 Adding support for TYP_HALF would require a fairly substantial commitment from the CoreCLR community, Microsoft and .NET.  It may even require changes to the ECMA standards.

A baby step would be to add intrinsics for the assembly instructions which do support Half precision floating point.  There are very limited support for Half in the base ARM64 standard.  Narrowing and Widening are two such operations.